jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsSubsystemDefinition.java

```diff
@@ -78,6 +79,7 @@ public void registerAdditionalRuntimePackages(ManagementResourceRegistration res
                     RuntimePackageDependency.required(JSON_API.getName()),
                     RuntimePackageDependency.optional(RESTEASY_ATOM.getName()),
                     RuntimePackageDependency.optional(RESTEASY_JAXRS.getName()),
+                    RuntimePackageDependency.optional(RESTEASY_CLIENT_MICROPROFILE.getName()),
```

Comment:
@asoldano @bstansberry , in case we want to create a microprofile-rest-client galleon layer, this dependency should be passive. It would mean that when provisioning the jaxrs layer, microprofile-rest-client would be no more provisioned (no org.jboss.resteasy.resteasy-client-microprofile nor org.eclipse.microprofile.restclient modules).
The only way to provision microprofile-rest-client support would be by provisioning the "microprofile-rest-client" layer that would depend on jaxrs/cdi/microprofile-config. This would have an impact on jaxrs existing layer and layers that depend on it.
This is really an UX improvement (ability to pick and choose mp spec), the gain in term of size is small (100k).

Reply:
@jfdenise Is there any reason this kind of adjustment needs to happen now, e.g. some incompatibility later?

AIUI the goal of this PR is not adding a new layer, so changes to support a new layer can come with that work.

Reply:
@bstansberry , this can happen later.